shell/pages/c/_cluster/uiplugins/index.vue

```diff
@@ -0,0 +1,606 @@
+          });
+
+          this.updatePluginInstallStatus(plugin.name, false);
+        }
+      });
+    },
+
+  },
+  methods:    {
+    filterChanged(f) {
+      this.view = f.selectedName;
+    },
+
+    // Developer Load is in the action menu
+    showDeveloperLoaddDialog() {
+      this.$refs.developerInstallDialog.showDialog();
+    },
+
+    showInstallDialog(plugin, ev) {
+      ev.target?.blur();
+      ev.preventDefault();
+      ev.stopPropagation();
+
+      this.$refs.installDialog.showDialog(plugin);
+    },
+
+    showUninstallDialog(plugin, ev) {
+      ev.target?.blur();
+      ev.preventDefault();
+      ev.stopPropagation();
+
+      this.$refs.uninstallDialog.showDialog(plugin);
+    },
+
+    didUninstall(plugin) {
+      if (plugin) {
+        this.updatePluginInstallStatus(plugin.name, 'uninstall');
+
+        // Clear the load error, if there was one
+        this.$store.dispatch('uiplugins/setError', { name: plugin.name, error: false });
+      }
+    },
+
+    didInstall(plugin) {
+      if (plugin) {
+        // Change the view to installed if we started installing a plugin
+        this.$refs.tabs?.select('installed');
+      }
+    },
+
+    showPluginDetail(plugin) {
+      this.$refs.infoPanel.show(plugin);
+    },
+
+    updatePluginInstallStatus(name, status) {
+      // console.log(`UPDATING PLUGIN STATUS: ${ name } ${ status }`);
+      Vue.set(this.installing, name, status);
+    },
+
+    setMenu(event) {
+      this.menuOpen = !!event;
+
+      if (event) {
+        this.menuTargetElement = this.$refs.actions;
+        this.menuTargetEvent = event;
+      } else {
+        this.menuTargetElement = undefined;
+        this.menuTargetEvent = undefined;
+      }
+    }
+  }
+};
+</script>
+
+<template>
+  <div class="plugins">
+    <div class="plugin-header">
+      <h2>{{ t('plugins.title') }}</h2>
+      <button
+        v-if="hasMenuActions"
+        ref="actions"
+        aria-haspopup="true"
+        type="button"
+        class="btn actions"
+        @click="setMenu"
+      >
+        <i class="icon icon-actions" />
+      </button>
+      <ActionMenu
+        v-if="hasMenuActions"
+        :custom-actions="menuActions"
+        :open="menuOpen"
+        :use-custom-target-element="true"
+        :custom-target-element="menuTargetElement"
+        :custom-target-event="menuTargetEvent"
+        @close="setMenu(false)"
+        @devLoad="showDeveloperLoaddDialog"
+      />
+    </div>
+
+    <PluginInfoPanel ref="infoPanel" />
+
+    <Tabbed ref="tabs" :tabs-only="true" @changed="filterChanged">
+      <Tab name="installed" label-key="plugins.tabs.installed" :weight="20" />
+      <Tab name="available" label-key="plugins.tabs.available" :weight="19" />
+      <Tab name="updates" label-key="plugins.tabs.updates" :weight="18" :badge="updates.length" />
+      <Tab name="all" label-key="plugins.tabs.all" :weight="17" />
+    </Tabbed>
+    <div v-if="loading" class="data-loading">
```

Comment:
This probably looks nicer, but is a departure from the standard `Loading` component

Reply:
We can review